india_compliance/gst_india/doctype/purchase_reconciliation_tool/exporter.py

```diff
@@ -0,0 +1,409 @@
+                    "move_range": True,
+                },
+            ]
+
+        # To add header data to the sheet
+        exporter.append_data(
+            worksheet,
+            data_header,
+            add_row=add_row,
+            add_column=1,
+            is_common_header=True,
+            column_widths=column_widths,
+            bg_color=header_color,
+            horizontal="center",
+            vertical="center",
+        )
+
+        # To add data to the sheet
+        exporter.append_data(
+            worksheet,
+            data,
+            add_row=6,
+            add_column=1,
+            is_common_header=False,
+            column_widths=column_widths,
+            bg_color=body_color,
+            wrap_text=False,
+        )
+
+        rule = exporter.highlight_cell(bg_color="e6b9b8", color="000000", text="Diff")
+        worksheet.conditional_formatting.add("A1:Y10", rule)
+
+        # To add additional background color to the sheet
+        exporter.add_background_color(worksheet, additional_style)
+
+    # if args.get("supplier_summary"):
+    #     supplier_summary = args.supplier_summary[1:]
+    #     supplier_summary_header = args.supplier_summary[0]
+    #     exporter.add_worksheet(
+    #         data=supplier_summary,
+    #         header=supplier_summary_header,
+    #         sheet_name=args.sheet_names[1],
+    #         position=1,
+    #         range="A2:F2",
+    #         header_color="d9d9d9",
+    #         body_color="f2f2f2",
+    #         row=2,
+    #         workbook=workbook,
+    #     )
+
+    # if args.get("data"):
+    #     invoice_summary = args.data[1:]
+    #     inv_main_header = args.data[0][0]
+    #     inv_sub_header = args.data[0][1]
+
+    #     # Invoice Summary Sheet
+    #     ws1 = workbook.create_sheet(
+    #         args.sheet_names[2], 2 if args.get("supplier_summary") else 1
+    #     )
+
+    #     if inv_main_header:
+    #         exporter.update_header_data(ws1, inv_main_header, 1, 8, 9)
+    #         exporter.setup_header_for_invoice_data(
+    #             ws1, font_size=12, range="H1:P2", color="c6d9f1"
+    #         )
+
+    #     if inv_sub_header:
+    #         exporter.update_header_data(ws1, inv_sub_header, 2, 1, 1)
+    #         exporter.setup_header_for_invoice_data(
+    #             ws1, font_size=12, range="Q1:Y2", color="d7e4bd"
+    #         )
+
+    #     if args.get("data"):
+    #         ws1.merge_cells("H1:P1")
+    #         ws1.merge_cells("Q1:Y1")
+
+    #     exporter.setup_header_for_invoice_data(ws1, range="A2:E2", color="d9d9d9")
+    #     exporter.setup_header_for_invoice_data(ws1, range="F2:G2", color="e6b9b8")
+    #     exporter.setup_header_for_invoice_data(ws1, range="A2:Y2")
+
+    #     exporter.update_data(ws1, invoice_summary, inv_sub_header, color="f2f2f2")
+
+    exporter.remove_worksheet(workbook, "Sheet")
+    exporter.save_workbook(workbook, f"{args.get('file_name')}.xlsx")
```

Comment:
Final Notes (Suggestive):
One Exporter class: pure openpyxl extension that can be used at any other place.
Data Prep and Summary Class: Define columns styles, data etc. Segregate functions for different sheets. Don't club all headers in one function etc.
Whitelisted function: Just initializes and exports the data.